### ISSUE TITLE
LazyModel - Static cache is causing problems in appservers

### ISSUE THREAD

**Issue by gabor-farkas** [wicketstuff/core#534]
We've created jboss/wildfly modules for our applications that contain the most common dependencies, including wicket, so that they don't have to be included every time with the deployment.

However, when including lazymodel with that wildfly module, after an application redeploy, we receive exceptions:

```
java.lang.ClassCastException: Cannot cast my.Class$$EnhancerLAZY$$c703f20 to my.Class
     at java.lang.Class.cast(Class.java:3369)
     at org.wicketstuff.lazymodel.reflect.Evaluation.proxy(Evaluation.java:139)
     at org.wicketstuff.lazymodel.LazyModel.from(LazyModel.java:538)
```

This boils down to the fact that the proxyfactory is always cached (`new CachingProxyFactory(new DefaultProxyFactory());`), and that thus the ConcurrentHashMap used by the CachingProxyFactory is static to the Evaluation class.
This way, this cache 'survives' application restarts, but the application classes are new on each application start.
I'm not sure though why that old enhancer is found from the cache for the new application model class.

**Comment by svenmeier** [comment]
We have to make the class cache application-specific, as PropertyResolver is doing it. Via an IInitializer we can clean up the cache when an application is destroyed.

**Comment by svenmeier** [comment]
All caching is now done scoped to the application. Could you test with a snapshot whether this fixes the problem?

**Comment by gabor-farkas** [comment]
The same exception still occurs, for me it looks like ```private static IProxyFactory proxyFactory``` still causes problem on the Evaluation class.

(Though in the meantime we found that we might face other diffuclties while trying to use wicket and related stuff as modules. Wicket-core itself uses a static map for wicket application names, so we have to keep all our wicket-filters with different names in web.xml-s)

**Comment by svenmeier** [comment]
Ah yes, IProxyFactory has another one of these static caches :(.

Thanks for giving it a try, I'll fix that one too asap.

**Comment by gabor-farkas** [comment]
cool, thanks, no urge from my side

**Comment by svenmeier** [comment]
Fixed with 7.6.0
